wagtail/wagtailadmin/api/serializers.py

```diff
@@ -51,15 +52,36 @@ def get_attribute(self, instance):
         return instance
 
     def to_representation(self, page):
+        # We require this weird param for the explorer.. hmm there might be an easier
+        # way do to it?
+        children_with_children = page.get_children().filter(Q(numchild__gt=0)).count()
+
         return OrderedDict([
             ('count', page.numchild),
             ('listing_url', get_model_listing_url(self.context, Page) + '?child_of=' + str(page.id)),
+            ('children_with_children', children_with_children)
```

Comment:
Perhaps a more generic way of implementing this would be to add descendants to the API as well and in the frontend we just check if the counts are the same?